Visualize elevator edges and vertices in Debug UI

hjvuor opened the pull request:
### Summary

- This PR adds a layer group called Elevators to the debug UI that contains instances of ElevatorVertex and ElevatorHopEdge
- To properly draw ElevatorHopEdges in case their start and end points are not at the same location, this adds a getGeometry method where the geometry is computed from the start and end vertices of the edge.

### Issue

This is related to improving indoor navigation #6829 

### Unit tests

style.json was regenerated by DebugStyleSpecTest.java

### Bumping the serialization version id

Minor changes made to both ElevatorHopEdge and ElevatorVertex which requires to bump serialization id

optionsome commented on application/src/main/java/org/opentripplanner/apis/vectortiles/DebugStyleSpec.java:
```diff
@@ -260,6 +263,35 @@ private static List<StyleBuilder> vertices(VectorSourceLayer vertices) {
 
+  private static List<StyleBuilder> elevators(VectorSourceLayer edges, VectorSourceLayer vertices) {
+    return List.of(
+      StyleBuilder.ofId("elevator-hop-edge")
+        .group(ELEVATORS_GROUP)
+        .typeLine()
+        .vectorSourceLayer(edges)
+        .edgeFilter(ElevatorHopEdge.class)
+        .lineColor(ORANGE)
+        .lineWidth(LINE_HALF_WIDTH)
+        .lineOffset(LINE_OFFSET)
+        .minZoom(6)
+        .maxZoom(MAX_ZOOM)
+        .intiallyHidden(),
```
These are slightly problematic as they are used for all elevators, but only visible for the horizontal elevators currently since the line has 0 length usually. I was thinking should we rename this so it's more clear that these will only show up for the horizontal elevators or somehow draw them as circles when they have 0 length and as lines when they don't. However, it's also possible that we don't do anything about it now.

optionsome commented on application/src/main/java/org/opentripplanner/inspector/vector/edge/EdgePropertyMapper.java:
```diff
@@ -26,6 +27,12 @@ protected Collection<KeyValue> map(Edge input) {
           kv("duration", e.getDuration().map(d -> d.toString()).orElse(null))
         );
+        case ElevatorHopEdge e -> List.of(
+          kv("permission", e.getPermission()),
```
Do we ever have different permissions for elevators? I was thinking that are they always walk only? In that case, this is probably not relevant information.

hjvuor replied:
The permission can be PEDESTRIAN or PEDESTRIAN_AND_BICYCLE depending on whether a bike is allowed into the elevator

hjvuor replied:
I also think that we want the debug ui to be as representative of internal state as possible because if something exists as a class property we may potentially need it for debugging

hjvuor commented:
I've modified this PR to add a getGeometryForDebugUi method with a default implementation in the Edge class. This way I can implement the special case for ElevatorHopEdge without the risk of breaking something by implementing getGeometry. In case the elevator edge has no horizontal component, I've also added logic to move the other end a little to make it visible in the UI. I'm not sure if that's the cleanest way to achieve what we want but I'd like to hear some opinions.

hjvuor commented on application/src/test/java/org/opentripplanner/graph_builder/module/osm/moduletests/ElevatorTest.java:
```diff
@@ -60,7 +61,7 @@ void testMultilevelNodeDuration() {
     assertThat(edges).hasSize(2);
     for (var edge : edges) {
-      assertEquals(62, edge.getTravelTime());
+      assertEquals(Duration.ofSeconds(62), edge.getTravelTime().orElse(null));
```
Thanks! I've fixed it
